### PR TITLE
OmsAgent: Stop Install in case of stopOnMultipleConnections with SCOM connection

### PR DESCRIPTION
**Pull request by lagalbra**
Implemented loose list_workspaces method in detect_multiple_connections for new install case
Restart omsagent after enable and reorganize some code
Restarting will be necessary while some solutions are not multi-homing-enabled and all processes are started during install

This version of the extension will be released as 1.4.45.3, containing OMSAgent bundle version 1.4.1-45.

### CODE REVIEW

File: OmsAgent/Fairfax/omsagent_ff.py [hunk 1]
```diff
@@ -430,6 +428,196 @@ def check_workspace_id_and_key(workspace_id, workspace_key):
...
+                   'machine will get billed multiple times for each ' \
+                   'workspace it report to. ' \
+                   '(LINUXOMSAGENTEXTENSION_ERROR_MULTIPLECONNECTIONS)')
+        # This exception will get caught by the main method
+        raise OmsAgentUnwantedMultipleConnectionsException(err_msg)
+    else:
+        detect_scom_connection
+
+
+def detect_scom_connection():
```
Comment (NarineM) [inline OmsAgent/Fairfax/omsagent_ff.py]: Seems very long to the function. Can you break it into multiple helper functions instead? Like "detect_omsadminpath", "detect_omiconfigeditor_path",etc.. and call them from detect_scom_connection()?

File: OmsAgent/Fairfax/omsagent_ff.py [hunk 2]
```diff
@@ -37,18 +37,29 @@
...
-BundleFileName = 'omsagent-1.4.0-45.universal.x64.sh'
-HUtilObject = None
+BundleFileName = 'omsagent-1.4.1-45.universal.x64.sh'
+GUIDRegex = r'[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}'
+GUIDOnlyRegex = r'^' + GUIDRegex + '$'
+
+# Paths
+OmsAdminPath = '/opt/microsoft/omsagent/bin/omsadmin.sh'
+OmsAgentServiceScript = '/opt/microsoft/omsagent/bin/service_control'
+OMIConfigEditorPath = '/opt/omi/bin/omiconfigeditor'
```
Comment (NarineM) [inline OmsAgent/Fairfax/omsagent_ff.py]: Please have the global variable naming convention consistent. Either be "OMSAdminPath" or change "OMIConfigEditorPath" be "OmiConfigEditorPath".

File: OmsAgent/Fairfax/omsagent_ff.py [hunk 3]
```diff
@@ -430,6 +428,196 @@ def check_workspace_id_and_key(workspace_id, workspace_key):
...
+                scom_port_open = True
+            elif exit_code is 1:
+                hutil_log_info('According to {0}, SCOM port is not ' \
+                               'open'.format(omsadmin_cmd))
+                return
+
+    # If omsadmin.sh option is not available, use omiconfigeditor
+    if (scom_port_open is None and os.path.exists(OMIConfigEditorPath)
+            and os.path.exists(OMIServerConfPath)):
+        omi_cmd = '{0} httpsport -q 1270 < {1}'.format(OMIConfigEditorPath,
```
Comment (NarineM) [inline OmsAgent/Fairfax/omsagent_ff.py]: Move hardcoded 1270 port number to be in global variables.

File: OmsAgent/Fairfax/omsagent_ff.py [hunk 4]
```diff
@@ -430,6 +428,196 @@ def check_workspace_id_and_key(workspace_id, workspace_key):
...
+            hutil_log_info('SCOM port is not listed in ' \
+                               '{0}'.format(OMIServerConfPath))
+            return
+    else:
+        hutil_log_info('SCOM port could not be determined to be open')
+        return
+
+    # Parse the certificate to determine if SCOM issued it
+    if os.path.exists(SCOMCertPath):
+        cert_cmd = 'openssl x509 -in {0} -noout -text'.format(SCOMCertPath)
```
Comment (NarineM) [inline OmsAgent/Fairfax/omsagent_ff.py]: Check if openssl is installed and have a proper shutdown when the client tool is not found.

File: OmsAgent/Fairfax/omsagent_ff.py [hunk 5]
```diff
@@ -430,6 +428,196 @@ def check_workspace_id_and_key(workspace_id, workspace_key):
...
+        cert_exit_code, cert_output = run_get_output(cert_cmd, chk_err = False,
+                                                     log_cmd = False)
+        if cert_exit_code is 0:
+            issuer_search = r'^[\s]*Issuer:[\s]*CN=SCX-Certificate/title=SCX' + GUIDRegex + ', DC=.*$'
```
Comment (NarineM) [inline OmsAgent/Fairfax/omsagent_ff.py]: Move issuer search with cert subject line and title be a global variable.

Comment (lagalbra) [reply OmsAgent/Fairfax/omsagent_ff.py]: Done

File: OmsAgent/Fairfax/omsagent_ff.py (hunk 4 above)
Comment (lagalbra) [reply OmsAgent/Fairfax/omsagent_ff.py]: Done

File: OmsAgent/Fairfax/omsagent_ff.py (hunk 3 above)
Comment (lagalbra) [reply OmsAgent/Fairfax/omsagent_ff.py]: Done

File: OmsAgent/Fairfax/omsagent_ff.py (hunk 1 above)
Comment (lagalbra) [reply OmsAgent/Fairfax/omsagent_ff.py]: Done

File: OmsAgent/Fairfax/omsagent_ff.py (hunk 2 above)
Comment (lagalbra) [reply OmsAgent/Fairfax/omsagent_ff.py]: Done

**Comment by lagalbra** [comment]
Rebased against master and fixed one typo

**Pull request merged**